### PR TITLE
adds reproc package

### PR DESCRIPTION
**Pull request by paul-reilly**
reproc package for "starting, stopping and communicating with external programs".

### CODE REVIEW

File: packages/r/reproc/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    on_install(function(package)
+        local configs = {}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "RelWithDebInfo"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DREPROC++=" .. (package:config("c++") and "ON" or "OFF"))
+        import("package.tools.cmake").install(package, configs, { buildir = "build" })
+    end)
+
+    on_test(function(package)
+        assert(package:has_cincludes("reproc/run.h"))
```
Comment (waruqi) [inline packages/r/reproc/xmake.lua:23]: please use has_cfuncs to check api and link

Comment (paul-reilly) [reply packages/r/reproc/xmake.lua]: No probs, that should be it now.

File: packages/r/reproc/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    add_deps("cmake")
+
+    add_configs("c++", { description = "Build reproc C++ library.", default = true, type = "boolean" })
+
+    on_install(function(package)
+        local configs = {}
+        table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:debug() and "Debug" or "RelWithDebInfo"))
+        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DREPROC++=" .. (package:config("c++") and "ON" or "OFF"))
```
Comment (waruqi) [inline packages/r/reproc/xmake.lua:18]: `++=`? does cmake support it?

Comment (paul-reilly) [reply packages/r/reproc/xmake.lua:18]: Apparently so: 

https://github.com/DaanDeMeyer/reproc#user

Comment (waruqi) [reply packages/r/reproc/xmake.lua:18]: ok

Comment (paul-reilly) [reply packages/r/reproc/xmake.lua:24]: I've added pthread and ws2_32 syslinks.

File: packages/r/reproc/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    set_homepage("https://github.com/DaanDeMeyer/reproc")
+    set_license("MIT")
+
+    add_urls("https://github.com/DaanDeMeyer/reproc.git")
+    add_versions("v14.2.4", "8aea3aef64c37a7ed0db759a7128a80c2540c808")
+
+    add_deps("cmake")
+
+    if is_plat("linux") or is_plat("mingw") then
+      add_syslinks("pthread")
```
Comment (waruqi) [inline packages/r/reproc/xmake.lua:13]: incorrect indent

Comment (paul-reilly) [reply packages/r/reproc/xmake.lua:13]: Oh shizzle, I'll wait for these checks to complete first and then push a quick fix.

File: packages/r/reproc/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package("reproc")
+
+    set_description("a cross-platform C/C++ library that simplifies starting, stopping and communicating with external programs.")
+    set_homepage("https://github.com/DaanDeMeyer/reproc")
+    set_license("MIT")
+
+    add_urls("https://github.com/DaanDeMeyer/reproc.git")
```
Comment (waruqi) [inline packages/r/reproc/xmake.lua:7]: please add tar.gz url `https://github.com/DaanDeMeyer/reproc/archive/refs/tags/$(version).tar.gz` and it's sha256 to add_versions

Comment (paul-reilly) [reply packages/r/reproc/xmake.lua]: Consider it done! :)

**Pull request merged**